pkg/common/http.go

```diff
@@ -59,15 +59,18 @@ func HTTPPRequest(httpType string, uri string, jsonData []byte, authorizationTok
 
 func HTTPResponseBody(httpResponse *http.Response) ([]byte, error) {
 	// Pull out response body
-	defer httpResponse.Body.Close()
-	httpResponseBodyBytes, err := io.ReadAll(httpResponse.Body)
-	if err != nil {
-		return nil, errors.Wrapf(err, "reading http response body failed")
+	var httpResponseBodyBytes []byte
+	if httpResponse != nil && httpResponse.Body != nil {
+		defer httpResponse.Body.Close()
+		limitSize := httpResponse.ContentLength
+		const mb134 = 1 << 27 //134MB

```

Comment:
Can you come up with a more descriptive name or add a comment that explains why it's 134MB?

Reply:
Then we should make a comment that it's arbitrary so that in the future no one has to guess why.